src/extension/extensionHandler.ts

```diff
@@ -25,6 +25,10 @@ export class ExtensionHandler<TMock> {
       return extensionOrPropertyName(this._mock);
     }
 
+    if (!maybePropertyHandler) {
```

Comment:
Considering that function overload it's not helping us enough (In my opinion people should not be able to call this method without that parameter unless they disable type checking. What we could do instead is to create a different method in the ExtensionHandler. @Pmyl Tell me If I am wrong :)

Reply:
I presume making the property handler (`maybePropertyHandler`) required is the desired behavior in this case?

Reply:
it should be required but because is overloading it's not. I am happy with the error. Something similar to the createMock one. "It looks like you are trying to get an extension for {propertyName} without specifying the handler.